### PR TITLE
[SYCL] Turn on -fsycl-id-queries-fit-in-int by default

### PR DESCRIPTION
**Pull request by smanna12**
This patch enables sycl_id_queries_fit_in_int macro to true on default mode to fix regressions and performance issue.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/include/clang/Driver/Options.td [hunk 1]
```diff
@@ -2489,7 +2489,7 @@ def fsycl_instrument_device_code : Flag<["-"], "fsycl-instrument-device-code">,
   HelpText<"Add ITT instrumentation intrinsics calls">,
   MarshallingInfoFlag<CodeGenOpts<"SPIRITTAnnotations">>;
 defm sycl_id_queries_fit_in_int: BoolFOption<"sycl-id-queries-fit-in-int",
-  LangOpts<"SYCLValueFitInMaxInt">, DefaultFalse,
+  LangOpts<"SYCLValueFitInMaxInt">, DefaultTrue,
```
Comment (erichkeane) [inline clang/include/clang/Driver/Options.td:2492]: I'm not up on the options rewrite, is there a '-fno-...' flag?

Comment (smanna12) [reply clang/include/clang/Driver/Options.td:2492]: yes, there is fno flag: -fno-sycl-id-queries-fit-in-int

**Review by erichkeane — APPROVED** [review]
Seems ok to me.

File: clang/include/clang/Driver/Options.td (hunk 1 above)
Comment (mdtoguchi) [reply clang/include/clang/Driver/Options.td:2492]: BoolFOption covers both variants (`-fsycl-id-queries-fit-in-int` and `-fno-sycl-id-queries-fit-in-it`)

**Review by mdtoguchi — APPROVED** [review]

**Comment by smanna12** [comment]
Thanks for the reviews.

 I am seeing 35 ESIMD tests fail with the patch. 

http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLin%2FLLVM_Test_Suite/detail/LLVM_Test_Suite/3023/pipeline

**Review by AaronBallman — APPROVED** [review]
LGTM aside from precommit failures.

**Comment by smanna12** [comment]
> LGTM aside from precommit failures.

@DenisBakhvalov mentions that, it's because of the llvm.assume intrinsic in device code. BE compiler cannot handle them. So, we cannot enable _SYCL_ID_QUERIES_FIT_IN_INT_ for ESIMD.

**Comment by DenisBakhvalov** [comment]
> > LGTM aside from precommit failures.
> 
> @DenisBakhvalov mentions that, it's because of the llvm.assume intrinsic in device code. BE compiler cannot handle them. So, we cannot enable _SYCL_ID_QUERIES_FIT_IN_INT_ for ESIMD.

Yes, and I'm not sure we can introduce any ESIMD-specific workaround (like we did previously) since we unified the SYCL-ESIMD compilation. I think the way to proceed here is to have vector GPU backend (used by ESIMD code) accept `llvm.assume` intrinsic. Unfortunately, this means that the current PR is blocked. Tagging @kbobrovs .

**Comment by kbobrovs** [comment]
> > > LGTM aside from precommit failures.
> > 
> > 
> > @DenisBakhvalov mentions that, it's because of the llvm.assume intrinsic in device code. BE compiler cannot handle them. So, we cannot enable _SYCL_ID_QUERIES_FIT_IN_INT_ for ESIMD.
> 
> Yes, and I'm not sure we can introduce any ESIMD-specific workaround (like we did previously) since we unified the SYCL-ESIMD compilation. I think the way to proceed here is to have vector GPU backend (used by ESIMD code) accept `llvm.assume` intrinsic. Unfortunately, this means that the current PR is blocked. Tagging @kbobrovs .

ID as int is a big performance improvement for DPC++ in many cases, AFAIK. So, it would be good to have a w/a until ESIMD BE is ready to accepts these. It should be safe w/a to remove llvm.assume in ESIMD kernels for now. I will create a PR shortly.

**Comment by kbobrovs** [comment]
> I will create a PR shortly.

https://github.com/intel/llvm/pull/3459

**Comment by smanna12** [comment]
> > I will create a PR shortly.
> 
> #3459

Thanks @kbobrovs for the  w/a.

**Comment by kbobrovs** [comment]
@smanna12, please rebase to latest llvm/sycl and rerun, ESIMD failures should be gone

**Comment by smanna12** [comment]
> @smanna12, please rebase to latest llvm/sycl and rerun, ESIMD failures should be gone

Thanks @kbobrovs for the quick fix.

**Comment by smanna12** [comment]
All ESIMD tests are passing now but this patch is causing regressions for SubGroup tests
Failed Tests (2):

SYCL :: SubGroup/sub_group_as.cpp
SYCL :: SubGroup/sub_group_as_vec.cpp

Unexpected result 0101 vs 0100
http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLin%2FLLVM_Test_Suite/detail/LLVM_Test_Suite/3132/pipeline

**Comment by smanna12** [comment]
Rebasing and testing the PR after https://github.com/intel/llvm-test-suite/pull/220 merging.

**Comment by smanna12** [comment]
@bader All pre-commit test failures are fixed. I did not make any change to any of the files on this PR. I am not sure why buildbot/Lit_with_Cuda is failing with lit WARNING: The PID of hang binary not found. This worked fine couple of days ago with the same PR.

**Comment by bader** [comment]
> @bader All pre-commit test failures are fixed.
> 
> I did not make any change to any of the files on this PR.
> 
> I am not sure why buildbot/Lit_with_Cuda is failing with warning like this:
> lit WARNING: The PID of hang binary not found.
> 
> This worked fine couple of days ago with same patch. Is this expected?

@tfzhu, is the machine in a broken state?

**Comment by smanna12** [comment]
> > @bader All pre-commit test failures are fixed.
> > I did not make any change to any of the files on this PR.
> > I am not sure why buildbot/Lit_with_Cuda is failing with warning like this:
> > lit WARNING: The PID of hang binary not found.
> > This worked fine couple of days ago with same patch. Is this expected?
> 
> @tfzhu, is the machine in a broken state?

Thanks @bader and @tfzhu for fixing the problem.

**Pull request merged**